web/src/types/option.ts

```diff
@@ -0,0 +1,199 @@
+/* eslint-disable unicorn/prevent-abbreviations */
+import { ToString } from './utils';
+import { Err, Ok, Result } from './result';
+
+export const Some = <S extends ToString>(value: S) => {
+  return new Option<S>(false, value);
+};
+
+export const None = () => {
+  return new Option<never>(true);
+};
+
+export class Option<S extends ToString> {
```

Comment:
Do we really need this `ToString` bound here?

Reply:
It was needed for the `okOr` and `okOrElse` methods, because the `Ok` value was part of the error message when calling `unwrapErr`. I removed the value from the error message (in bad24a5) and thus we don't need this bound anymore.